# Analyse de base des fichiers concernant les lycées

In [1]:
# import bibliothéques d'analyse de python
import pandas as pd
import matplotlib.pyplot as plt
from matplotlib.ticker import MaxNLocator
from matplotlib.ticker import FuncFormatter
import seaborn as sns
import numpy as np

import fonctions.perso_stats as perso_stats

In [2]:
ips_lycee = pd.read_csv('donnee/fr-en-ips-lycees-ap2022.csv',sep=';')
va_lycee_gen = pd.read_csv('donnee/fr-en-indicateurs-de-resultat-des-lycees-denseignement-general-et-technologique.csv',sep=';',low_memory=False)
va_lycee_pro = pd.read_csv('donnee/fr-en-indicateurs-de-resultat-des-lycees-denseignement-professionnels.csv',sep=';',low_memory=False)

In [3]:
ips_lycee.head(1)

rentree_scolaire academie code_du_departement departement       uai  \
0        2022-2023     LYON                 001         AIN  0010021T   

                               nom_de_l_etablissment code_insee_de_la_commune  \
0  LYCEE PROFESSIONNEL MARCELLE PARDE LYCEE DES M...                    01053   

  nom_de_la_commune secteur type_de_lycee  effectifs_voie_gt  \
0   BOURG EN BRESSE  public            LP                NaN   

   effectifs_voie_pro  effectifs_ensemble_gt_pro  ips_voie_gt  ips_voie_pro  \
0               520.0                      520.0          NaN          86.5   

   ips_ensemble_gt_pro  ecart_type_de_l_ips_voie_gt  \
0                 86.5                          NaN   

   ecart_type_de_l_ips_voie_pro  
0                          24.3

In [4]:
ips_lycee.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3598 entries, 0 to 3597
Data columns (total 18 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   rentree_scolaire              3598 non-null   object 
 1   academie                      3598 non-null   object 
 2   code_du_departement           3598 non-null   object 
 3   departement                   3598 non-null   object 
 4   uai                           3598 non-null   object 
 5   nom_de_l_etablissment         3598 non-null   object 
 6   code_insee_de_la_commune      3598 non-null   object 
 7   nom_de_la_commune             3598 non-null   object 
 8   secteur                       3598 non-null   object 
 9   type_de_lycee                 3598 non-null   object 
 10  effectifs_voie_gt             2464 non-null   float64
 11  effectifs_voie_pro            2075 non-null   float64
 12  effectifs_ensemble_gt_pro     3598 non-null   float64
 13  ips

Le fichier fr-en-ips-lycees-ap2022.csv contient 18 colonnes<br>
<ul>
  <li>10 colonnes pour decriere le lycée (adresse, type de lycée,...)</li>
  <li>une colonne qui indique la rentrée au format XXXX-YYYY</li>
  <li>3 colonnes qui décrivent les effectifs général, professionnel et totaux</li>
  <li>5 colonnes pour décrire l'IPS selon le cursus général ou pro, ainsi que l'ecart-type</li>
</ul>

On va créer un carnet d'adresse avec le numero uai à partir de ce fichier.<br>
On va reformater la colonne rentrée scolaire (passée de XXXX-YYYY à XXXX)

In [5]:
va_lycee_gen.head(1)

etablissement  annee   ville code_etablissement  \
0  LYCEE XAVIER BICHAT (GENERAL ET TECHNO.)   2021  NANTUA           0010032E   

  commune academie departement secteur_public_pu_prive_pr  \
0   01269     LYON         AIN                     public   

   effectif_presents_serie_l  effectif_presents_serie_es  ...  \
0                        NaN                         NaN  ...   

   taux_men_brut_gnle  va_men_gnle  \
0                70.0           -7   

   nombre_de_mentions_tb_avec_felicitations_g  \
0                                         NaN   

   nombre_de_mentions_tb_sans_felicitations_g  nombre_de_mentions_b_g  \
0                                         NaN                     NaN   

   nombre_de_mentions_ab_g  nombre_de_mentions_tb_avec_felicitations_t  \
0                      NaN                                         NaN   

   nombre_de_mentions_tb_sans_felicitations_t  nombre_de_mentions_b_t  \
0                                         NaN                     NaN   

   nombre_de_mentions_ab_t  
0                      NaN  

[1 rows x 145 columns]

In [6]:
va_lycee_gen.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27808 entries, 0 to 27807
Columns: 145 entries, etablissement to nombre_de_mentions_ab_t
dtypes: float64(80), int64(3), object(62)
memory usage: 30.8+ MB


Le fichier fr-en-indicateurs-de-resultat-des-lycees-denseignement-general-et-technologique.csv contient les 145 colonnes suivantes :

In [7]:
for a in va_lycee_gen.columns :
    print(a)

etablissement
annee
ville
code_etablissement
commune
academie
departement
secteur_public_pu_prive_pr
effectif_presents_serie_l
effectif_presents_serie_es
effectif_presents_serie_s
effectif_presents_serie_stg
effectif_presents_serie_sti2d
effectif_presents_serie_std2a
effectif_presents_serie_stmg
effectif_presents_serie_sti
effectif_presents_serie_stl
effectif_presents_serie_st2s
effectif_presents_serie_musiq_danse
effectif_presents_serie_hotellerie
effectif_presents_total_series
taux_brut_de_reussite_serie_l
taux_brut_de_reussite_serie_es
taux_brut_de_reussite_serie_s
taux_brut_de_reussite_serie_stg
taux_brut_de_reussite_serie_sti2d
taux_brut_de_reussite_serie_std2a
taux_brut_de_reussite_serie_stmg
taux_brut_de_reussite_serie_sti
taux_brut_de_reussite_serie_stl
taux_brut_de_reussite_serie_st2s
taux_brut_de_reussite_serie_musiq_danse
taux_brut_de_reussite_serie_hotellerie
taux_brut_de_reussite_total_series
taux_reussite_attendu_acad_serie_l
taux_reussite_attendu_acad_serie_es
taux_reuss

Le fichier fr-en-indicateurs-de-resultat-des-lycees-denseignement-general-et-technologique.csv contient 145 colonnes<br>
<ul>
  <li>une colonne annee qui est l'année de la rentré scolaire</li>
  <li>11 colonne pour décrire l'etablissement (adresse, secteur public ou privé, numéro uai)</li>
  <li>pour chaque serie:
    <ul>
      <li>une colonne effectif_present qui donne le nombre de canditats qui pqsse le bac</li>
      <li>les taux brut de réussite</li>
      <li>les taux de réussite attendu en france, dans l'academie</li>
      <li>les taux de mention brut et attendu</li>
      <li>la valeur ajoutée pour le taux de réussite et la valeur ajouté pour le taux mention</li>
    </ul>
    </li>
    <li>8 colonnes qui donne le nombre de mention par college toute serie confondu</li>
</ul>
Alors pour limiter le nombre de colonnes : création d'une colonne serie et fusion de toutes les colonnes des series.<br>


In [8]:
va_lycee_pro.head(1)

etablissement code_etablissement  annee  \
0  LYCEE PROFESSIONNEL SAINTE JEANNE D'ARC           0592928M   2019   

              ville commune academie departement secteur_public_1_prive_2  \
0  AULNOYE AYMERIES   59033    LILLE        NORD       privé sous contrat   

   effectifs_presents_specialites_pluri_technologiques  \
0                                                NaN     

   effectifs_presents_transformations  ...  va_men_pluri_serv  \
0                                 5.0  ...                NaN   

   va_men_echanges  va_men_communication  va_men_serv_pers  \
0              NaN                   NaN                 3   

   va_men_serv_collec  va_men_services  va_men_total  \
0                 NaN                3           5.0   

   nombre_de_mentions_tb_sans_felicitations_p  nombre_de_mentions_b_p  \
0                                         NaN                     NaN   

   nombre_de_mentions_ab_p  
0                      NaN  

[1 rows x 130 columns]

In [9]:
va_lycee_pro.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24236 entries, 0 to 24235
Columns: 130 entries, etablissement to nombre_de_mentions_ab_p
dtypes: float64(62), int64(3), object(65)
memory usage: 24.0+ MB


Le fichier fr-en-indicateurs-de-resultat-des-lycees-denseignement-professionnels.csv contient les 130 colonnes suivantes :

In [10]:
for a in va_lycee_pro.columns :
    print(a)

etablissement
code_etablissement
annee
ville
commune
academie
departement
secteur_public_1_prive_2
effectifs_presents_specialites_pluri_technologiques
effectifs_presents_transformations
effectifs_presents_genie_civil_construction_et_bois
effectifs_presents_materiaux_souples
effectifs_presents_mecanique_electricite_electronique
effectif_presents_production
effectifs_presents_specialites_plurivalentes_des_services
effectifs_presents_echanges_et_gestion
effectifs_presents_communication_et_information
effectifs_presents_services_aux_personnes
effectifs_presents_services_a_la_collectivite
effectif_presents_services
effectif_presents_total_secteurs
taux_brut_de_reussite_specialites_pluri_technologiques
taux_brut_de_reussite_transformations
taux_brut_de_reussite_genie_civil_construction_et_bois
taux_brut_de_reussite_materiaux_souples
taux_brut_de_reussite_mecanique_electricite_electronique
taux_brut_de_reussite_production
taux_brut_de_reussite_specialites_plurivalentes_des_services
taux_brut_

Le fichier contient les éléments suivants :<br>
<ul>
  <li>Une colonne annee avec l'année scolaire</li>
  <li>9 colonnes qui décrivent le lycée (adresse, secteur public ou privé, uai, ...)</li>
  <li>3 colonnes pour les mentions (pas de mention très bien avec félicitation par rapport au lycée général)</li>
  <li>pour chaque spécialité professionnelle :
    <ul>
      <li>une colonne effectif qui dinne le nombre de candidat diplome preparé</li>
      <li>un taux brut de réussite</li>
      <li>un taux de réussite attendu en france</li>
      <li>un taux de réussite attendu dans l'academie</li>
      <li>un taux brut de mention</li>
      <li>un taux de mention attendu</li>
      <li>la valeur ajouté du à l'etablissement au taux de réussite</li>
      <li>la valeur ajouté du à l établissement au taux de mention</li>
      <li>la valeur ajouté du à l'etablissement au taux d'accès (le lycée garde ces élèves au cours de la formation)</li>
    </ul>
  </li>
  <li>les taux d'accès de l'etablissement</li>
</ul>

Pour l'instant, je me concentre sur les différentes series, et l'IPS des établissements.<br>
Pour cela je vais créer comme pour le lycée général une colonne serie et fussionner les différente colonnes communes.<br>

In [11]:
adresse_lycee = ips_lycee[['uai','academie','departement','code_du_departement','nom_de_l_etablissment','code_insee_de_la_commune','nom_de_la_commune','type_de_lycee']].groupby(by='uai',group_keys=False).apply(lambda x :x)

In [12]:
ips_lycee['rentree_scolaire'].unique()

array(['2022-2023'], dtype=object)

In [13]:
ips_lycee['rentree_scolaire'] = ips_lycee['rentree_scolaire'].str[:4].astype(int)

In [14]:
ips_lycee.drop(columns=['secteur','academie',
'code_du_departement','departement','nom_de_l_etablissment',
'code_insee_de_la_commune','nom_de_la_commune','type_de_lycee',
'effectifs_voie_gt','effectifs_voie_pro','effectifs_ensemble_gt_pro'],inplace=True)

In [15]:
ips_lycee.rename(columns={'rentree_scolaire':'rentree'},inplace=True)

In [16]:
va_lycee_gen.rename(columns={'code_etablissement':'uai','annee':'rentree','secteur_public_pu_prive_pr':'secteur'},inplace=True)

In [17]:
# enlevé les resneignement type adresse
va_lycee_gen.drop(columns=['commune','ville','academie','departement','etablissement'],inplace=True)

In [18]:
for a in va_lycee_gen.columns :
    print(a)

rentree
uai
secteur
effectif_presents_serie_l
effectif_presents_serie_es
effectif_presents_serie_s
effectif_presents_serie_stg
effectif_presents_serie_sti2d
effectif_presents_serie_std2a
effectif_presents_serie_stmg
effectif_presents_serie_sti
effectif_presents_serie_stl
effectif_presents_serie_st2s
effectif_presents_serie_musiq_danse
effectif_presents_serie_hotellerie
effectif_presents_total_series
taux_brut_de_reussite_serie_l
taux_brut_de_reussite_serie_es
taux_brut_de_reussite_serie_s
taux_brut_de_reussite_serie_stg
taux_brut_de_reussite_serie_sti2d
taux_brut_de_reussite_serie_std2a
taux_brut_de_reussite_serie_stmg
taux_brut_de_reussite_serie_sti
taux_brut_de_reussite_serie_stl
taux_brut_de_reussite_serie_st2s
taux_brut_de_reussite_serie_musiq_danse
taux_brut_de_reussite_serie_hotellerie
taux_brut_de_reussite_total_series
taux_reussite_attendu_acad_serie_l
taux_reussite_attendu_acad_serie_es
taux_reussite_attendu_acad_serie_s
taux_reussite_attendu_acad_serie_stg
taux_reussite_atten

In [19]:
va_lycee_gen['rentree'].unique()

array([2021, 2020, 2019, 2018, 2017, 2016, 2015, 2014, 2013, 2012, 2022,
       2023])

In [20]:
va_lycee_gen.head()

rentree       uai             secteur  effectif_presents_serie_l  \
0     2021  0010032E              public                        NaN   
1     2021  0010075B  privé sous contrat                        NaN   
2     2021  0011119L              public                        NaN   
3     2021  0011276G              public                        NaN   
4     2021  0020032Z              public                        NaN   

   effectif_presents_serie_es  effectif_presents_serie_s  \
0                         NaN                        NaN   
1                         NaN                        NaN   
2                         NaN                        NaN   
3                         NaN                        NaN   
4                         NaN                        NaN   

   effectif_presents_serie_stg  effectif_presents_serie_sti2d  \
0                          NaN                            NaN   
1                          NaN                            NaN   
2                          NaN                           68.0   
3                          NaN                           36.0   
4                          NaN                            NaN   

   effectif_presents_serie_std2a  effectif_presents_serie_stmg  ...  \
0                            NaN                           NaN  ...   
1                            NaN                          12.0  ...   
2                           33.0                           NaN  ...   
3                            NaN                          89.0  ...   
4                            NaN                          69.0  ...   

   taux_men_brut_gnle  va_men_gnle  \
0                70.0           -7   
1                84.0            1   
2                71.0           -8   
3                72.0          -10   
4                73.0           -1   

   nombre_de_mentions_tb_avec_felicitations_g  \
0                                         NaN   
1                                         NaN   
2                                         NaN   
3                                         NaN   
4                                         NaN   

   nombre_de_mentions_tb_sans_felicitations_g  nombre_de_mentions_b_g  \
0                                         NaN                     NaN   
1                                         NaN                     NaN   
2                                         NaN                     NaN   
3                                         NaN                     NaN   
4                                         NaN                     NaN   

   nombre_de_mentions_ab_g  nombre_de_mentions_tb_avec_felicitations_t  \
0                      NaN                                         NaN   
1                      NaN                                         NaN   
2                      NaN                                         NaN   
3                      NaN                                         NaN   
4                      NaN                                         NaN   

   nombre_de_mentions_tb_sans_felicitations_t  nombre_de_mentions_b_t  \
0                                         NaN                     NaN   
1                                         NaN                     NaN   
2                                         NaN                     NaN   
3                                         NaN                     NaN   
4                                         NaN                     NaN   

   nombre_de_mentions_ab_t  
0                      NaN  
1                      NaN  
2                      NaN  
3                      NaN  
4                      NaN  

[5 rows x 140 columns]

In [21]:
# faire une liste des series dans le fichier va_lycee_gen
liste_serie_gen = []

for nom_colonne in va_lycee_gen.columns:
    if nom_colonne.startswith('effectif_presents_serie'):
        liste_serie_gen.append(nom_colonne[18:])

In [22]:
# faire un dataframe avec serie

lycee_gen = pd.DataFrame(columns=['rentree','uai','secteur','serie','type_diplome_gen_pro','effectif_present'
                                    ,'taux_brut_de_reussite','taux_attendu_serie_academie','taux_attendu_serie_fr'
                                    ,'taux_mention_brut','taux_mention_attendu_serie'])
for serie in  liste_serie_gen: 
    liste_colonne = []
    colonne_mention =[]
    dico_serie = {'effectif_presents_'+serie:'effectif_present'
                ,'taux_brut_de_reussite_'+serie:'taux_brut_de_reussite'
                ,'taux_reussite_attendu_acad_'+serie:'taux_attendu_serie_academie'
                ,'taux_reussite_attendu_france_'+serie:'taux_attendu_serie_fr'
                ,'taux_mention_brut_'+serie:'taux_mention_brut'
                ,'taux_mention_attendu_'+serie:'taux_mention_attendu_serie'
                }

    for colonne in va_lycee_gen.columns:
        if colonne.endswith(serie):
            liste_colonne.append(colonne)
        if colonne.startswith('nombre_de_mentions'):
            colonne_mention.append(colonne)
    liste_colonne +=['rentree','secteur','uai']
    temps_data = va_lycee_gen[~va_lycee_gen['effectif_presents_'+serie].isna()][liste_colonne].copy()
    temps_data['serie']= serie
    temps_data['type_diplome_gen_pro'] = 'gen'
    temps_data.rename(columns=dico_serie,inplace=True)
    lycee_gen = pd.concat([lycee_gen,temps_data])


In [23]:
lycee_gen.info()

<class 'pandas.core.frame.DataFrame'>
Index: 85861 entries, 468 to 27725
Data columns (total 11 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   rentree                      85861 non-null  object 
 1   uai                          85861 non-null  object 
 2   secteur                      85861 non-null  object 
 3   serie                        85861 non-null  object 
 4   type_diplome_gen_pro         85861 non-null  object 
 5   effectif_present             85861 non-null  float64
 6   taux_brut_de_reussite        85861 non-null  float64
 7   taux_attendu_serie_academie  24760 non-null  float64
 8   taux_attendu_serie_fr        66891 non-null  object 
 9   taux_mention_brut            43359 non-null  float64
 10  taux_mention_attendu_serie   25522 non-null  object 
dtypes: float64(4), object(7)
memory usage: 7.9+ MB


In [24]:
lycee_gen.tail()

rentree       uai             secteur             serie  \
27586    2022  0622807P              public  serie_hotellerie   
27587    2022  0623902E              public  serie_hotellerie   
27691    2022  0771658J              public  serie_hotellerie   
27700    2022  0781578S              public  serie_hotellerie   
27725    2022  0850130E  privé sous contrat  serie_hotellerie   

      type_diplome_gen_pro  effectif_present  taux_brut_de_reussite  \
27586                  gen              36.0                  100.0   
27587                  gen              20.0                   85.0   
27691                  gen              10.0                  100.0   
27700                  gen              65.0                  100.0   
27725                  gen              15.0                  100.0   

       taux_attendu_serie_academie taux_attendu_serie_fr  taux_mention_brut  \
27586                          NaN                   NaN               69.0   
27587                          NaN                   NaN               55.0   
27691                          NaN                   NaN               40.0   
27700                          NaN                   NaN               62.0   
27725                          NaN                   NaN               67.0   

      taux_mention_attendu_serie  
27586                        NaN  
27587                        NaN  
27691                        NaN  
27700                        NaN  
27725                        NaN

In [25]:
va_lycee_pro.rename(columns={'code_etablissement':'uai','annee':'rentree','secteur_public_1_prive_2':'secteur'},inplace=True)

In [26]:
va_lycee_pro.drop(columns=['etablissement','ville','commune','academie','departement'],inplace=True)

In [27]:
va_lycee_pro.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24236 entries, 0 to 24235
Columns: 125 entries, uai to nombre_de_mentions_ab_p
dtypes: float64(62), int64(3), object(60)
memory usage: 23.1+ MB


In [28]:
for a in va_lycee_pro.columns :
    print(a)

uai
rentree
secteur
effectifs_presents_specialites_pluri_technologiques
effectifs_presents_transformations
effectifs_presents_genie_civil_construction_et_bois
effectifs_presents_materiaux_souples
effectifs_presents_mecanique_electricite_electronique
effectif_presents_production
effectifs_presents_specialites_plurivalentes_des_services
effectifs_presents_echanges_et_gestion
effectifs_presents_communication_et_information
effectifs_presents_services_aux_personnes
effectifs_presents_services_a_la_collectivite
effectif_presents_services
effectif_presents_total_secteurs
taux_brut_de_reussite_specialites_pluri_technologiques
taux_brut_de_reussite_transformations
taux_brut_de_reussite_genie_civil_construction_et_bois
taux_brut_de_reussite_materiaux_souples
taux_brut_de_reussite_mecanique_electricite_electronique
taux_brut_de_reussite_production
taux_brut_de_reussite_specialites_plurivalentes_des_services
taux_brut_de_reussite_echanges_et_gestion
taux_brut_de_reussite_communication_et_informat

In [29]:
va_lycee_pro.head()

uai  rentree             secteur  \
0  0592928M     2019  privé sous contrat   
1  0592963A     2019  privé sous contrat   
2  0592967E     2019  privé sous contrat   
3  0592976P     2019  privé sous contrat   
4  0592980U     2019  privé sous contrat   

   effectifs_presents_specialites_pluri_technologiques  \
0                                                NaN     
1                                                NaN     
2                                                NaN     
3                                                NaN     
4                                                NaN     

   effectifs_presents_transformations  \
0                                 5.0   
1                                 NaN   
2                                 NaN   
3                                 NaN   
4                                 NaN   

   effectifs_presents_genie_civil_construction_et_bois  \
0                                                NaN     
1                                                NaN     
2                                                NaN     
3                                                NaN     
4                                                NaN     

   effectifs_presents_materiaux_souples  \
0                                   NaN   
1                                   NaN   
2                                   NaN   
3                                   NaN   
4                                   NaN   

   effectifs_presents_mecanique_electricite_electronique  \
0                                                NaN       
1                                                NaN       
2                                                NaN       
3                                                NaN       
4                                                NaN       

   effectif_presents_production  \
0                           5.0   
1                           NaN   
2                           NaN   
3                           NaN   
4                           NaN   

   effectifs_presents_specialites_plurivalentes_des_services  ...  \
0                                                NaN          ...   
1                                                NaN          ...   
2                                               22.0          ...   
3                                                7.0          ...   
4                                               14.0          ...   

   va_men_pluri_serv  va_men_echanges  va_men_communication  va_men_serv_pers  \
0                NaN              NaN                   NaN                 3   
1                NaN              NaN                   NaN               -11   
2                 -7               -6                   NaN                10   
3                -21               -5                   NaN                -2   
4                 -8              NaN                   NaN                 2   

   va_men_serv_collec  va_men_services  va_men_total  \
0                 NaN                3           5.0   
1                 NaN              -11         -11.0   
2                 NaN                1           1.0   
3                 NaN               -5          -5.0   
4                 NaN                0           0.0   

   nombre_de_mentions_tb_sans_felicitations_p  nombre_de_mentions_b_p  \
0                                         NaN                     NaN   
1                                         NaN                     NaN   
2                                         NaN                     NaN   
3                                         NaN                     NaN   
4                                         NaN                     NaN   

   nombre_de_mentions_ab_p  
0                      NaN  
1                      NaN  
2                      NaN  
3                      NaN  
4                      NaN  

[5 rows x 125 columns]

In [30]:
#listez les serie générale et technologique
liste_serie_pro = []
for nom_colonne in va_lycee_pro.columns:
    if nom_colonne.startswith('effectifs'):
        liste_serie_pro.append(nom_colonne[19:])


In [31]:
# faire un dataframe avec serie

lycee_pro = pd.DataFrame(columns=['rentree','uai','secteur','serie','type_diplome_gen_pro','effectif_present'
                                    ,'taux_brut_de_reussite','taux_attendu_serie_academie','taux_attendu_serie_fr'
                                    ,'taux_mention_brut','taux_mention_attendu_serie'])
for serie in  liste_serie_pro: 
    liste_colonne = []
    dico_serie = {'effectifs_presents_'+serie:'effectif_present'
                ,'taux_brut_de_reussite_'+serie:'taux_brut_de_reussite'
                ,'taux_reussite_attendu_acad_'+serie:'taux_attendu_serie_academie'
                ,'taux_de_reussite_attendu_france_'+serie:'taux_attendu_serie_fr'
                ,'taux_brut_de_mentions_'+serie:'taux_mention_brut'
                ,'taux_de_mentions_attendu_'+serie:'taux_mention_attendu_serie'
                }
    for colonne in va_lycee_pro.columns:
        if colonne.endswith(serie):
            liste_colonne.append(colonne)
    liste_colonne +=['rentree','secteur','uai'
                    ]
    temps_data = va_lycee_pro[~va_lycee_pro['effectifs_presents_'+serie].isna()][liste_colonne].copy()
    temps_data['serie']= serie
    temps_data['type_diplome_gen_pro'] = 'pro'
    temps_data.rename(columns=dico_serie,inplace=True)
    lycee_pro = pd.concat([lycee_pro,temps_data])

In [32]:
lycee_pro

rentree       uai             secteur                             serie  \
36       2019  0630061Z              public  specialites_pluri_technologiques   
74       2019  0750712S              public  specialites_pluri_technologiques   
119      2019  0840015K              public  specialites_pluri_technologiques   
120      2019  0840042P              public  specialites_pluri_technologiques   
191      2019  9740479C              public  specialites_pluri_technologiques   
...       ...       ...                 ...                               ...   
24193    2018  0610692W  privé sous contrat        services_a_la_collectivite   
24200    2018  0620150B              public        services_a_la_collectivite   
24221    2018  0670024M              public        services_a_la_collectivite   
24233    2018  0762976X              public        services_a_la_collectivite   
24234    2018  0763002A              public        services_a_la_collectivite   

      type_diplome_gen_pro  effectif_present  taux_brut_de_reussite  \
36                     pro              15.0                   87.0   
74                     pro              15.0                   87.0   
119                    pro              11.0                  100.0   
120                    pro               9.0                  100.0   
191                    pro              20.0                   90.0   
...                    ...               ...                    ...   
24193                  pro              15.0                  100.0   
24200                  pro              11.0                   91.0   
24221                  pro              10.0                   70.0   
24233                  pro              12.0                   75.0   
24234                  pro              14.0                   71.0   

      taux_attendu_serie_academie taux_attendu_serie_fr  taux_mention_brut  \
36                            NaN                    85               53.0   
74                            NaN                    74               47.0   
119                           NaN                    75               64.0   
120                           NaN                    86               56.0   
191                           NaN                    80               70.0   
...                           ...                   ...                ...   
24193                         NaN                    96               67.0   
24200                         NaN                    86                9.0   
24221                         NaN                    85                0.0   
24233                         NaN                    82               25.0   
24234                         NaN                    84               36.0   

      taux_mention_attendu_serie  
36                            44  
74                            29  
119                           34  
120                           46  
191                           41  
...                          ...  
24193                         62  
24200                         33  
24221                         30  
24233                         28  
24234                         27  

[43941 rows x 11 columns]

In [33]:
#merger les fichiers ensembles
fichier_lycee = pd.concat([lycee_gen,lycee_pro])
fichier_lycee = pd.merge(fichier_lycee,ips_lycee, on=['uai','rentree'],how='left')
fichier_lycee = pd.merge(fichier_lycee,adresse_lycee, on='uai',how='left')
#pas mal de colonne voir pour diminuer celle-ci ou pas


In [34]:
fichier_lycee.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 129802 entries, 0 to 129801
Data columns (total 23 columns):
 #   Column                        Non-Null Count   Dtype  
---  ------                        --------------   -----  
 0   rentree                       129802 non-null  object 
 1   uai                           129802 non-null  object 
 2   secteur                       129802 non-null  object 
 3   serie                         129802 non-null  object 
 4   type_diplome_gen_pro          129802 non-null  object 
 5   effectif_present              129802 non-null  float64
 6   taux_brut_de_reussite         129802 non-null  float64
 7   taux_attendu_serie_academie   24760 non-null   float64
 8   taux_attendu_serie_fr         91009 non-null   object 
 9   taux_mention_brut             75862 non-null   float64
 10  taux_mention_attendu_serie    40000 non-null   object 
 11  ips_voie_gt                   4865 non-null    float64
 12  ips_voie_pro                  6193 non-null 

In [35]:
fichier_lycee['secteur'].replace({'public':'PUBLIC','privé sous contrat':'PRIVE'},inplace=True)

In [36]:
fichier_lycee.to_csv('donnee_temp/lycee.csv',index=False)